README.md

```diff
@@ -19,141 +20,189 @@ When the reader has completed this Code Pattern, they will understand how to:
+Use your existing machine learning providor
 
-Create the following services:
+<details><summary>Setup on a VM</summary>
 
-* [Watson OpenScale](https://cloud.ibm.com/catalog/services/ai-openscale)
-  You will get the Watson OpenScale instance GUID when you run the notebook using the [IBM Cloud CLI](https://cloud.ibm.com/catalog/services/ai-openscale)
+### Custom Machine Learning Provider Setup for VM
 
-* [Databases for PostgreSQL DB](https://cloud.ibm.com/catalog/services/databases-for-postgresql)
+The code in flask.py can be used to start a gunicorn/flask application that can be hosted in a VM, such that it can be accessable from CPD system.
+This code does the following:
+* It wraps a Watson Machine Learning model that is deployed to a space.
+* The hosting application URL should contain the SPACE ID and the DEPLOYMENT ID. The app can be used to talk to the target WML model/deployment.
+* Having said that, this is only for this tutorial purpose, and you can define your Custom ML provider endpoint in any fashion you want, such that it wraps your own custom ML engine.
+* The scoring request and response payload should confirm to the schema as described here at: https://dataplatform.cloud.ibm.com/docs/content/wsj/model/wos-frameworks-custom.html
+* To start the application using the below code, make sure you install the python packages using:
 
+```python
 
-* Wait a couple of minutes for the database to be provisioned.
-* Click on the `Service Credentials` tab on the left and then click `New credential +` to create the service credentials. Copy them or leave the tab open to use later in the notebook.
+pip install -r requirement.txt
+```
 
-### 3. Create a notebook in IBM Watson Studio
+</details>
 
-* In [Watson Studio](https://dataplatform.cloud.ibm.com/), create a `New project`.
-* Using the project you've created, click on `+ Add to project` and then choose the  `Notebook` tile, OR in the `Assets` tab under `Notebooks` choose `+ New notebook` to create a notebook.
-* Select the `From URL` tab.
-* Enter a name for the notebook.
-* Optionally, enter a description for the notebook.
-* Under `Notebook URL` provide the following url: https://raw.githubusercontent.com/IBM/monitor-custom-ml-engine-with-watson-openscale/master/notebooks/WatsonOpenScaleAndCustomMLEngine.ipynb
-* Select the `Default Python 3.5` runtime, either `Free` or `XS`.
-* Click the `Create` button.
+or 
 
-### 4a. Deploy to IBM Cloud
+<details><summary>Run on RedHat OpenShift</summary>
 
-[![Deploy to IBM Cloud](https://cloud.ibm.com/devops/setup/deploy/button.png)](https://cloud.ibm.com/devops/setup/deploy?repository=https://github.com/IBM/monitor-custom-ml-engine-with-watson-openscale)
+## Prerequisites
 
-Click the ``Deploy to IBM Cloud`` button and hit ``Create`` and then jump to step 5.
+You will need a running OpenShift cluster, or OKD cluster. You can provision [OpenShift on the IBM Cloud](https://cloud.ibm.com/kubernetes/catalog/openshiftcluster).
```

Comment:
OKD? should that be spelled out the first time?

Reply:
Actually, I'd like to narrow it to OpenShift only, since the instructions are specifc to that.